project_administrator_restricted_visibility/hooks.py

```diff
@@ -10,12 +10,15 @@ def uninstall_hook(cr, registry):
     # 'Project: Administrator' group (project.group_project_manager) to 'Administrator'
     # in order to avoid getting a SQL constraint error:
     # 'duplicate key value violates unique constraint "res_groups_name_uniq'"
-    group_full_project_manager = env.ref(
-        "project_administrator_restricted_visibility.group_full_project_manager"
+    module_name = "project_administrator_restricted_visibility"
+    group_full_project_manager = env.ref(f"{module_name}.group_full_project_manager")
+    rule_restricted_project_manager = env.ref(
+        f"{module_name}.project_project_restricted_manage_own_project_rule"
     )
     env.ref("project.project_project_manager_rule").write(
         {"groups": [(3, group_full_project_manager.id)]}
     )
+    rule_restricted_project_manager.unlink()
```

Comment:
issue: you don't really need to remove this record. The record is namespaced to its module, so Odoo will remove it automatically when uninstalling.

The purpose of these hooks is to undo changes on records that belong to other modules.

So, IUUC, all changes in this file could be removed.